Compiler/Script/Interactive.mo

```diff
@@ -10048,6 +10037,245 @@ algorithm
+end updateConnectionNames;
+
+protected function updateConnectionNamesInClass
+  "Helper function to updateConnectionNames."
+  input Absyn.Class inClass1;
+  input String inFrom;
+  input String inTo;
+  input String inFromNew;
+  input String inToNew;
+  output Absyn.Class outClass;
+algorithm
+  outClass:=
+  match (inClass1, inFrom, inTo, inFromNew, inToNew)
+    local
+      list<Absyn.EquationItem> eqlst,eqlst_1;
+      list<Absyn.ClassPart> parts2,parts;
+      String i, bcname;
+      Boolean p,f,e;
+      Absyn.Restriction r;
+      Option<String> cmt;
+      SourceInfo file_info;
+      list<Absyn.ElementArg> modif;
+      list<String> typeVars;
+      list<Absyn.NamedArg> classAttrs;
+      list<Absyn.Annotation> ann;
+      String from, to, fromNew, toNew;
+    /* a class with parts */
+    case (Absyn.CLASS(name = i,partialPrefix = p,finalPrefix = f,encapsulatedPrefix = e,restriction = r,
+                      body = Absyn.PARTS(typeVars = typeVars,classAttrs = classAttrs,classParts = parts,ann=ann,comment = cmt),info = file_info),from,to,fromNew,toNew)
+      equation
+        eqlst = getEquationList(parts);
+        eqlst_1 = updateConnectionNamesInEqlist(eqlst, from, to, fromNew, toNew);
+        parts2 = replaceEquationList(parts, eqlst_1);
+      then
+        Absyn.CLASS(i,p,f,e,r,Absyn.PARTS(typeVars,classAttrs,parts2,ann,cmt),file_info);
+    /* an extended class with parts: model extends M end M;  */
+    case (Absyn.CLASS(name = i,partialPrefix = p,finalPrefix = f,encapsulatedPrefix = e,restriction = r,
+                      body = Absyn.CLASS_EXTENDS(baseClassName = bcname,modifications=modif,parts = parts,ann = ann,comment = cmt),info = file_info),from,to,fromNew,toNew)
+      equation
+        eqlst = getEquationList(parts);
+        eqlst_1 = updateConnectionNamesInEqlist(eqlst, from, to, fromNew, toNew);
+        parts2 = replaceEquationList(parts, eqlst_1);
+      then
+        Absyn.CLASS(i,p,f,e,r,Absyn.CLASS_EXTENDS(bcname,modif,cmt,parts2,ann),file_info);
+  end match;
+end updateConnectionNamesInClass;
+
+protected function updateConnectionNamesInEqlist
+  "Helper function to updateConnectionNames."
+  input list<Absyn.EquationItem> inAbsynEquationItemLst1;
+  input String inFrom;
+  input String inTo;
+  input String inFromNew;
+  input String inToNew;
+  output list<Absyn.EquationItem> outAbsynEquationItemLst;
+algorithm
+  outAbsynEquationItemLst := match (inAbsynEquationItemLst1,inFrom,inTo,inFromNew,inToNew)
+    local
+      list<Absyn.EquationItem> res,xs;
+      Absyn.ComponentRef c1,c2;
+      Absyn.Info info1;
+      Option<Absyn.Comment> cmt;
+      Absyn.EquationItem x;
+      String from, to, fromNew, toNew;
+
+    case ({},_,_,_,_) then {};
+    case (((x as Absyn.EQUATIONITEM(equation_ = Absyn.EQ_CONNECT(connector1 = c1,connector2 = c2), comment = cmt, info = info1)) :: xs),from,to,fromNew,toNew)
+      guard
+        stringEq(Absyn.crefString(c1),from) and stringEq(Absyn.crefString(c2), to)
```

Comment:
same as above `or stringEq(Absyn.crefString(c1), to) and stringEq(Absyn.crefString(c2), from)`

Reply:
This should preferably also be rewritten in a more modern style.